frontend/src/components/useUsers.js

```diff
@@ -24,15 +28,20 @@ const useUsers = (searchKeyword, sortKey, isSortAscending) => {
     fetchUsers()
   }, [])
 
-  const sendRegistrationApproval = useCallback(async (user) => {
-    const registrationStatus = await sendApproval(user)
+  const sendRegistrationApproval = useCallback(async (userId) => {
+    const registrationStatus = await sendApproval(userId)
 
     if (registrationStatus.status === 200) {
       console.log('Fresh users list from BE.')
       fetchUsers()
     }
   }, [])
 
+  const deleteUser = useCallback(async (userId) => {
+    await deleteUserRegistration(userId)
+    fetchUsers()
```

Comment:
A fetchUsers()-t egy setTimeout-ban 3mp időzítéssel elhelyezve egész jó a UX, a modal kap 3mp játékidőt, de tudom ez nem ment volna át a szűrőn. :)

Reply:
A probléma gyökere, hogy a Modal a gomb komponensben van, így ha a gomb komponens eltűnik, akkor a modal is. A gombnak annyi a felelősségi köre, hogy szól, ha rákattintanak ezzel kifejezve a törlési szándékot. Utána az, hogy ezt meg lehet erősíteni vagy vissza lehet vonni már más dolog. Ezen kívül olyan sosem lesz, hogy egyszerre két törlés gombra kattint a felhasználó és egyszerre két modal-t akarunk megjeleníteni. Ebből én arra következtetek, hogy a modal-t és a confirmation logikát a lista komponensbe célszerű helyezni és megoldódik minden probléma.

Reply:
Az időzítéses trükközések általában valami bugot vagy tervezési hibát rejtenek el, ha ilyen megoldás merül föl akkor érdemes átgondolni, hogy mi a probléma gyökere. Erre egy jó technika a 5 whys, szintén LEAN és Toyota gyökerekből eredően: https://en.wikipedia.org/wiki/Five_whys

Reply:
Profin működik.